File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {

        int n = word1.length();
        int m = word2.length();

        int i = 0;
        int j = 0;
        int turn = 1;

        string res = "";

        while (i<n || j<m){
            turn = turn == 1?0:1;
            if(i<n && turn == 0){
                res += word1[i];
                i += 1;
            }
            else if(j<m){
                res += word2[j];
                j += 1;
            }
        }
        return res;

        
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:

        i = 0
        j = 0
        turn  = 1
        res =""

        while i<len(word1) or j<len(word2):
            turn = 0 if turn == 1 else 1
            if i<len(word1) and turn == 0:
                res += word1[i]
                i += 1
            elif j <len(word2):
                res += word2[j]
                j += 1
        return res